src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Recipes/ElasticsearchIndexStep.cs

```diff
@@ -26,20 +26,31 @@ ElasticsearchIndexManager elasticIndexManager
 
     protected override async Task HandleAsync(RecipeExecutionContext context)
     {
-        if (context.Step["Indices"] is not JsonArray jsonArray)
-        {
-            return;
-        }
+        // Elasticserach uses the term "Indices" for the plural of index, but the common spelling in US English is
+        // "Indexes". So both are supported to avoid unnecessary spelling problems.
+        var indexes = context.Step["Indexes"] as JsonArray ??
+                context.Step["Indices"] as JsonArray ??
+                [];
 
-        foreach (var index in jsonArray)
-        {
-            var elasticIndexSettings = index.ToObject<Dictionary<string, ElasticIndexSettings>>().FirstOrDefault();
+        // Get all properties of each objects inside the indexes array. The property name is treated as the index name.
+        var settings = indexes
+            .SelectMany(index => index.ToObject<Dictionary<string, ElasticIndexSettings>>())
+            .Select(pair => WithIndexName(pair.Value, pair.Key));
```

Comment:
```suggestion
            .Select(pair => WithIndexName);
```

Reply:
Try to check before applying this change

Reply:
You don't need the arrow part either, just pass the name of the method.
```suggestion
            .Select(WithIndexName);
```

Reply:
I think that

Reply:
For some reason the other suggestion didn't actually get committed. It works after I manually applied it.